tests/execution.rs

```diff
@@ -3453,12 +3453,47 @@ fn test_err_fixed_stack_out_of_bound() {
     );
 }
 
+#[test]
+fn test_execution_overrun() {
+    let config = Config {
+        enabled_sbpf_versions: SBPFVersion::V1..=SBPFVersion::V1,
+        ..Config::default()
+    };
+    test_interpreter_and_jit_asm!(
+        "
+        add r1, 0",
+        config.clone(),
+        [],
+        (),
+        TestContextObject::new(2),
+        ProgramResult::Err(EbpfError::ExecutionOverrun),
+    );
+    test_interpreter_and_jit_asm!(
+        "
+        add r1, 0",
```

Comment:
Why does this exceed maximum instructions while the lddw case below does not? Add is one instruction while lddw is two.

Reply:
Ah, Lddw accounts for a single instruction, not two. Let me reformulate my question. Why a does single instruction raise the ExceededMaxInstructions when we only request one unit for execution?

Reply:
`lddw` only costs one CU. And this tests `add`, not `lddw` here.

Reply:
> `lddw` only costs one CU.

I wrote exactly that in my previous comment:

> Ah, Lddw accounts for a single instruction, not two

There is no need to repeat what I said. You've probably missed my comment. Now, my question is: 

>  Why a does single instruction raise the ExceededMaxInstructions when we only request one unit for execution?

So in the `add` test, we only have a single add instruction and request one CU, but we raise an `ExceededMaxInstructions` error. Why is this the case?

> And this tests `add`, not `lddw` here.

I didn't understand how this answers my question.

Reply:
> So in the add test, we only have a single add instruction and request one CU, but we raise an ExceededMaxInstructions error. Why is this the case?

I suppose logically, the ExceededMaxInstructions and ExecutionOverrun faults happen after the same time here, and it's not clear what to return.
After `add` is executed, the virtual CPU is in a state where it expects another instruction. But at this point CU==0 and PC is OOB, so the next instruction would fault for either of these reasons.

I suppose you could specify that instruction fetch occurs before CU checks, but that's just going to inhibit future possible optimizations.